src/commands/Register.ts

```diff
@@ -1,149 +1,235 @@
-  const { app, contracts, blockchain } = getServerConfig();
-  let registerConfig: RegisterConfig | undefined = undefined;
-  if (config.has('register')) {
-    registerConfig = config.get('register');
+  const bal = await provider.getBalance(relayAddress);
+
+  if (bal.gt(options.funds)) {
+    log.info('Relayer already funded');
+  } else {
+    log.info(
+      `Funding relayer ${utils.formatUnits(options.funds, 'ether')} RBTC`
+    );
+
+    const fundTx = await options.signer.sendTransaction({
+      to: relayAddress,
+      value: options.funds,
+      gasLimit: 1e6,
+      gasPrice: options.gasPrice,
+    });
+
+    if (fundTx.hash == null) {
+      throw new Error(`Fund transaction reverted: ${JSON.stringify(fundTx)}`);
+    }
+    transactions.push(fundTx.hash);
   }
-  log.setLevel(app.logLevel);
-  const register = new Register(
-    blockchain.rskNodeUrl,
-    registerConfig?.mnemonic
-  );
-  const portIncluded: boolean = app.url.indexOf(':') > 0;
-  const relayUrl =
-    app.url + (!portIncluded && app.port > 0 ? ':' + app.port.toString() : '');
-  await register.execute(
-    registerOptions
-      ? registerOptions
-      : {
-          hub: contracts.relayHubAddress,
-          signer: await register.findWealthyAccount(),
-          stake: utils.parseEther(registerConfig?.stake ?? '0.01'),
-          funds: utils.parseEther(registerConfig?.funds ?? '0.02'),
-          relayUrl,
-          unstakeDelay: '1000',
-          gasPrice: '60000000',
-        }
+
+  await waitForRelay(httpClient, options.relayUrl);
+  log.info('Executed Transactions', transactions);
+};
+
+const retreiveSigner = async (
+  rpcProvider: JsonRpcProvider,
+  privateKey?: string,
+  mnemonic?: string
+) => {
+  const walletFromPK = privateKey && new Wallet(privateKey, rpcProvider);
+  const walletFromMnemonic =
+    mnemonic && Wallet.fromMnemonic(mnemonic).connect(rpcProvider);
+
+  return (
+    walletFromPK ||
+    walletFromMnemonic ||
+    (await findWealthyAccount(rpcProvider))
   );
-}
+};
+
+const executeRegister = async (): Promise<void> => {
+  const { app, contracts, blockchain } = getServerConfig();
+  log.setLevel(app.logLevel);
+  log.debug('configSources', config.util.getConfigSources());
+  if (!config.has('register')) {
+    throw new Error(
+      'No register config found. Make sure that the register section exists in default.json5.'
+    );
+  }
+
+  const {
+    stake,
+    funds,
+    mnemonic,
+    privateKey,
+    signer,
+    gasPrice,
+    hub,
+    relayUrl,
+    unstakeDelay,
+  }: RegisterConfig = config.get('register');
+
+  const rpcProvider = new JsonRpcProvider(blockchain.rskNodeUrl);
+  const portFromUrl = app.url.match(/:(\d{0,5})$/);
+  const serverUrl =
+    !portFromUrl && app.port ? `${app.url}:${app.port}` : app.url;
+
+  await register(rpcProvider, {
+    hub: hub ?? contracts.relayHubAddress,
+    relayUrl: relayUrl || serverUrl,
+    signer: signer._isSigner
```

Comment:
`signer` could be undefined and it would raise an error.
```suggestion
    signer: signer?._isSigner
```

Reply:
I actually removed that because in the default.json5 the signer is defined as  `{}`.
We should either verify all of the values on start-up or expect users to not change the default values and only overwrite to non-undefined

Reply:
Live discussion concluded we just use the default value, so no change is necessary here.

Reply:
done